includes/User.php

```diff
@@ -1141,6 +1155,23 @@ private function loadFromSession() {
 			return false;
 		}
 
+		if ( !$this->isUserAuthenticated() ) {
```

Comment:
I think we should do that only after we a) thoroughly understand how this works or b) implement a comprehensive security model and apply it to MW. In other words, I don't want to break any "security" measures that are currently in place.